crates/client/rpc/src/lib.rs

```diff
@@ -652,4 +652,113 @@ where
     ) -> RpcResult<DeclareTransactionResult> {
         todo!("Not implemented")
     }
+
+    /// Returns a transaction details from it's hash.
+    ///
+    /// It first verifies if the transaction was included
+    /// in an older block, verifying db mapping in the client database.
+    /// If not found, it also verifies if the transaction is pending
+    /// in the runtime.
+    ///
+    /// # Arguments
+    ///
+    /// * `transaction_hash` - Hex string of the transaction hash.
+    fn get_transaction_by_hash(&self, transaction_hash: FieldElement) -> RpcResult<Transaction> {
+        let block_hash_from_db = self
+            .backend
+            .mapping()
+            .block_hash_from_transaction_hash(H256::from(transaction_hash.to_bytes_be()))
+            .map_err(|e| {
+                error!("Failed to get transaction's substrate block hash from mapping_db: {e}");
+                StarknetRpcApiError::TxnHashNotFound
+            })?;
+
+        let substrate_block_hash = match block_hash_from_db {
+            Some(block_hash) => block_hash,
+            None => {
+                // TODO: verify if the tx is in the tx pool (self.pool.ready()).
+                return Err(StarknetRpcApiError::TxnHashNotFound.into());
+            }
+        };
+
+        let block = self
+            .overrides
+            .for_block_hash(self.client.as_ref(), substrate_block_hash)
+            .current_block(substrate_block_hash)
+            .unwrap_or_default();
+
+        match block.transactions() {
+            BlockTransactions::Full(transactions) => {
+                for (txn, _) in transactions {
+                    if txn.hash != transaction_hash.into() {
+                        continue;
+                    }
+
+                    return Ok(Transaction::try_from(txn.clone()).map_err(|e| {
+                        error!("{:?}", e);
+                        StarknetRpcApiError::InternalServerError
+                    })?);
+                }
```

Comment:
Using iterators and `find` is probably more idiomatic. Is there a reason why not to use it?

Reply:
I was limited by the fact that I didn't understand how to manage the possible error in the find. I have tried an implementation in the new commit with find.